The Mac runners of our CI do not present the same error as the linked reproducer. After a debugging session with @siliataider I realize it's because one fundamental ingredient to reproduce the issue is the Python version: the OP's environment as well as my machine have Python 3.14 . For reference, I paste here the result of the added ctest on my machine:

```
6: Test command: /Users/vpadulan/Programs/python-venvs/rootdev/bin/python3.14 "post_install_test_ACLiC.py"
6: Working Directory: /tmp/PostInstallBuild
6: Environment variables: 
6:  PYTHONPATH=/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib
6: Test timeout computed to be: 1500
6: Info in <TUnixSystem::ACLiC>: creating shared library /private/tmp/PostInstallBuild/post_install_test_ACLiC_C.so
6: ld: cannot link directly with 'libffi-trampolines.dylib' because product being built is not an allowed client of it
6: clang++: error: linker command failed with exit code 1 (use -v to see invocation)
6: Error in <ACLiC>: Executing 'cd "/private/tmp/PostInstallBuild" ; c++ -fPIC -c -g  -Wc++11-narrowing -Wsign-compare -Wsometimes-uninitialized -Wconditional-uninitialized -Wheader-guard -Warray-bounds -Wcomment -Wtautological-compare -Wstrncat-size -Wloop-analysis -Wbool-conversion -m64 -pipe -W -Woverloaded-virtual -fsigned-char -fsized-deallocation -fno-common -Qunused-arguments -pthread -stdlib=libc++ -std=c++20  -I/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/include -isysroot "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" -I"/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/etc/" -I"/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/etc//cling" -I"/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/etc//cling/plugins/include" -I"/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/include" -I"/opt/homebrew/include" -I"/opt/homebrew/opt/python@3.14/Frameworks/Python.framework/Versions/3.14/include/python3.14"   -D__ACLIC__ "/private/tmp/PostInstallBuild/post_install_test_ACLiC_C_ACLiC_dict.cxx" ; c++ -g  "/private/tmp/PostInstallBuild/post_install_test_ACLiC_C_ACLiC_dict.o" -dynamiclib -Wl,-w -Wl,-undefined,dynamic_lookup -Wl,-headerpad_max_install_names -m64 -Wl,-dead_strip_dylibs -Wl,-install_name,@rpath/post_install_test_ACLiC_C.so  -Wl,-rpath,/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib -Wl,-rpath,/private/tmp/PostInstallBuild  "/opt/homebrew/Cellar/python@3.14/3.14.3_1/Frameworks/Python.framework/Versions/3.14/Python" "/usr/lib/libffi-trampolines.dylib" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/cppyy/libcppyy.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libCPyCppyy.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libCore.so" "/opt/homebrew/Cellar/lz4/1.10.0/lib/liblz4.1.10.0.dylib" "/opt/homebrew/Cellar/zstd/1.5.7_1/lib/libzstd.1.5.7.dylib" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libRIO.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libThread.so" "/opt/homebrew/Cellar/tbb/2022.2.0/lib/libtbb.12.16.dylib" "/opt/homebrew/Cellar/tbb/2022.2.0/lib/libtbbmalloc.2.16.dylib" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libCling.so" "/opt/homebrew/Cellar/xz/5.8.2/lib/liblzma.5.dylib" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/ROOT/libROOTPythonizations.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libTree.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libImt.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libNet.so" "/Users/vpadulan/Programs/rootproject/rootbuild/gh-22235-distrdf-debug/lib/libMultiProc.so" -o "/private/tmp/PostInstallBuild/post_install_test_ACLiC_C.so" ' failed!
6: Traceback (most recent call last):
6:   File "/private/tmp/PostInstallBuild/post_install_test_ACLiC.py", line 21, in <module>
6:     raise SystemExit(main())
6:                      ~~~~^^
6:   File "/private/tmp/PostInstallBuild/post_install_test_ACLiC.py", line 12, in main
6:     raise RuntimeError(f"Failed to compile post_install_test_ACLiC.C, error code: {err.value}")
6: RuntimeError: Failed to compile post_install_test_ACLiC.C, error code: 1
1/2 Test #6: python-post-install-ACLiC ................***Failed    9.39 sec
test 7
    Start 7: python-post-install-ACLiC-cleanupfiles
```

Unfortunately, there is no easy way to run on our self-hosted Mac runners with a different Python version than the system one, so we may have to accept this PR based on my local results alone. Or think about a way to run a ROOT installation from a PR build artifacts on a different MacOS runner with a different Python version.

--- a/.github/workflows/root-ci.yml
+++ b/.github/workflows/root-ci.yml
@@ -113,6 +113,7 @@ jobs:
     env:
       VIRTUAL_ENV_DIR: ${{ github.workspace }}/ROOT_CI_VENV
       MACOSX_DEPLOYMENT_TARGET: ${{ matrix.platform == 'mac-beta' && '27' || '' }} # Problems with gfortran falling back 18 (unsupported)
+      HOME: /Users/sftnight
 
     strategy:
       fail-fast: false
@@ -154,7 +155,6 @@ jobs:
       - name: Pull Request Build
         if: github.event_name == 'pull_request'
         env:
-          HOME: /Users/sftnight
           INCREMENTAL: ${{ !contains(github.event.pull_request.labels.*.name, 'clean build') && !matrix.platform == 'mac15' && !matrix.platform == 'mac26'}}
           GITHUB_PR_ORIGIN: ${{ github.event.pull_request.head.repo.clone_url }}
           OVERRIDES: ${{ join( matrix.overrides, ' ') }}
@@ -230,6 +230,30 @@ jobs:
           name: Binaries ${{ matrix.platform }} ${{ matrix.arch }}
           path: /Users/sftnight/ROOT-CI/packages/root_v*
           if-no-files-found: error
+
+      - name: Install
+        env:
+          BUILD_DIR: ${{ env.HOME }}/ROOT-CI/build
+          INSTALL_DIR: ${{ env.HOME }}/ROOT-CI/install
+        if: ${{ !cancelled() && !matrix.is_special }}
+        run: "cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_DIR }}"
+
+      - name: Build post-install test project
+        if: ${{ !cancelled() && !matrix.is_special }}
+        env:
+          SOURCE_DIR: ${{ env.HOME }}/ROOT-CI/src/test/PostInstall/
+          INSTALL_DIR: ${{ env.HOME }}/ROOT-CI/install
+          POST_INSTALL_DIR: ${{ env.HOME }}/ROOT-CI/PostInstall
+        run: |
+          cmake -S ${{ env.SOURCE_DIR }} -B ${{ env.POST_INSTALL_DIR }} -DCMAKE_PREFIX_PATH=${{ env.INSTALL_DIR }};
+          cmake --build ${{ env.POST_INSTALL_DIR }};
+
+      - name: CTest in post-install test project
+        if: ${{ !cancelled() && !matrix.is_special }}
+        env:
+          POST_INSTALL_DIR: ${{ env.HOME }}/ROOT-CI/PostInstall
+        working-directory: ${{ env.POST_INSTALL_DIR }}
+        run: ctest --output-on-failure -j $(sysctl -n hw.logicalcpu)
 
 
   build-windows:

--- a/core/base/src/TSystem.cxx
+++ b/core/base/src/TSystem.cxx
@@ -2276,6 +2276,11 @@ const char *TSystem::GetLibraries(const char *regexp, const char *options,
 
    static TRegexp separator("[^ \\t\\s]+");
    static TRegexp dynload("/lib-dynload/");
+   // Skip libffi, it is a private library used by the system. This is visible in the stub .tbd file:
+   // allowable-clients:
+   //     clients:         [ '!' ]
+   // See https://github.com/Homebrew/homebrew-core/issues/272324#issuecomment-5119880493 for more info
+   static TRegexp libffiMatch("/usr/lib/libffi");
 
    Ssiz_t start, index, end;
    start = index = end = 0;
@@ -2284,7 +2289,7 @@ const char *TSystem::GetLibraries(const char *regexp, const char *options,
       index = libs2.Index(separator, &end, start);
       if (index >= 0) {
          TString s = libs2(index, end);
-         if (s.Index(dynload) == kNPOS) {
+         if (s.Index(dynload) == kNPOS && s.Index(libffiMatch) == kNPOS) {
             if (!maclibs.IsNull()) maclibs.Append(" ");
             maclibs.Append(s);
          }

